static/js/components/nl_search_bar/auto_complete_input.tsx

```diff
@@ -0,0 +1,348 @@
+      latestQuery.current = currentText;
+      placeAutocompleteService.current.getPredictions(
+        {
+          input: latestQuery.current,
+          types: ["(regions)"],
+          offset: currentText.length,
+        },
+        (predictions, status) =>
+          onPlaceAutocompleteCompleted(
+            currentText,
+            predictions,
+            status,
+            /* allowRetry= */ true
+          )
+      );
+    }
+  }
+
+  function changeText(text: string) {
+    setInputText(text);
+    onChange(text);
+  }
+
+  // For all clicks outside of the input component, empty out results.
+  OutsideClickAlerter(wrapperRef, () => {
+    setResults({ placeResults: [], svResults: [] });
+  });
+
+  function handleKeydownEvent(
+    event: React.KeyboardEvent<HTMLDivElement>
+  ): void {
+    switch (event.key) {
+      case "Enter":
+        if (hoveredIdx >= 0) {
+          onClick(allResults[hoveredIdx]);
+        } else {
+          onSearch();
+        }
+        event.preventDefault();
+        break;
+      case "ArrowUp":
+        event.preventDefault();
+        processArrowKey(Math.max(hoveredIdx - 1, -1));
+        break;
+      case "ArrowDown":
+        event.preventDefault();
+        processArrowKey(Math.min(hoveredIdx + 1, allResults.length - 1));
+        break;
+    }
+  }
+
+  function processArrowKey(selectedIndex: number) {
+    setHoveredIdx(selectedIndex);
+    const textDisplayed =
+      selectedIndex >= 0
+        ? replaceQueryWithSelection(allResults[selectedIndex].name)
+        : baseInput;
+    changeText(textDisplayed);
+  }
+
+  return (
+    <>
+      <div className="search-box-section" ref={wrapperRef}>
+        <div className={`search-bar${value ? " non-empty" : ""}`}>
+          <InputGroup className="search-bar-content">
+            {isHeaderBar && (
+              <span className="material-icons-outlined">search</span>
+            )}
+            <Input
+              id={inputId}
+              invalid={invalid}
+              placeholder={placeholder}
+              value={inputText}
+              onChange={onInputChange}
+              onKeyDown={(event) => handleKeydownEvent(event)}
+              className="pac-target-input search-input-text"
+              autoComplete="one-time-code"
+              autoFocus={shouldAutoFocus}
+            ></Input>
+            <div onClick={onSearch} id="rich-search-button">
+              {isHeaderBar && (
+                <span className="material-icons-outlined">arrow_forward</span>
+              )}
+            </div>
+          </InputGroup>
+        </div>
+        {enableAutoComplete && (
+          <AutoCompleteSuggestions
+            inputText={inputText}
+            allResults={allResults}
+            hoveredIdx={hoveredIdx}
+            onClick={onClick}
+          />
+        )}
+      </div>
+    </>
+  );
+
+  function onClick(result) {
+    if (result.name.toLowerCase().includes(inputText.toLowerCase())) {
```

Comment:
So I changed a little to account for typo correction. But basically, it's meant to trigger a place explorer redirect on selection when the query that was sent to the google maps api is the entire result.

For example, "Population of Califor" --> Matched_query would be "Califor", so it will go to explore on select.
But "Angelo" --> Selecting "Angelópolis" will trigger place redirection